From e451957eb9f8e6bc2fe7c8eba11c3f1a54a321ed Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Fri, 4 Dec 2015 14:06:53 +0530
Subject: [PATCH 1/4] [EM/PMIC_CCSM]: Update charger type detection timeout
 based on status

As per the BC1.2 spec DCD detection should happen between 300msec and
900msec. When detecting the charger type based on the current status
(detection not started/failed or on going) adjusting the timeout to
complete the charge detection. Here 200msec timeout given for retry
if the charger type is not detected/failed/ongoing to complete the
detection with the retry count.

Change-Id: I2265ccb6b514068fe3ed620ddcae1c6b5f29eb67
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-12471
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
Reviewed-on: https://android.intel.com:443/444396
(cherry picked from commit bfa474467179a13ee468c019586c65839dcda3ff)
Reviewed-on: https://android.intel.com:443/447240
---
 drivers/power/intel_pmic_ccsm.c |   13 ++++++-------
 drivers/power/intel_pmic_ccsm.h |    5 ++---
 2 files changed, 8 insertions(+), 10 deletions(-)

Index: b/drivers/power/intel_pmic_ccsm.c
===================================================================
--- a/drivers/power/intel_pmic_ccsm.c	2016-03-08 13:36:14.516621330 -0800
+++ b/drivers/power/intel_pmic_ccsm.c	2016-03-08 13:38:14.678440249 -0800
@@ -845,6 +845,7 @@
 static int get_charger_type(void)
 {
 	int ret, i = 0;
+	int time = 0;
 	u8 val;
 	int chgr_type, rid;
 
@@ -858,18 +859,16 @@
 		if ((val & USBSRCDET_SUSBHWDET) ==
 			USBSRCDET_SUSBHWDET_DETSUCC)
 			break;
-		else {
-			pmic_write_reg(chc.reg_map->pmic_usbphyctrl,
-						USBPHYCTRL_CTYPE_START);
-			usleep_range(USBSRCDET_SLEEP_MIN_TIME,
-					USBSRCDET_SLEEP_MAX_TIME);
-		}
+		else
+			msleep(USBSRCDET_SLEEP_RETRYDET);
+
+		time += USBSRCDET_SLEEP_RETRYDET;
 	} while (i < USBSRCDET_RETRY_CNT);
 
 	if ((val & USBSRCDET_SUSBHWDET_DETSUCC) !=
 			USBSRCDET_SUSBHWDET_DETSUCC) {
 		dev_err(chc.dev, "Charger detection unsuccessful after %dms\n",
-			i * USBSRCDET_SLEEP_MIN_TIME);
+			time);
 		return 0;
 	}
 
Index: b/drivers/power/intel_pmic_ccsm.h
===================================================================
--- a/drivers/power/intel_pmic_ccsm.h	2016-03-08 13:36:14.516621330 -0800
+++ b/drivers/power/intel_pmic_ccsm.h	2016-03-08 13:36:41.153025268 -0800
@@ -149,12 +149,11 @@
 
 
 #define USBSRCDET_RETRY_CNT		5
-#define USBSRCDET_SLEEP_MIN_TIME	300000
-#define USBSRCDET_SLEEP_MAX_TIME	900000
+#define USBSRCDET_SLEEP_RETRYDET	200 /* 200mSec */
 #define USBSRCDET_SUSBHWDET_MASK	(3 << 0)
 #define USBSRCDET_USBSRCRSLT_MASK	(0x0F << 2)
 #define USBSRCDET_SDCD_MASK		(1 << 6)
-#define USBSRCDET_SUSBHWDET_DETON	(1 << 0)
+#define USBSRCDET_SUSBHWDET_INPROG	(1 << 0)
 #define USBSRCDET_SUSBHWDET_DETSUCC	(1 << 1)
 #define USBSRCDET_SUSBHWDET_DETFAIL	(3 << 0)
 #define USBSRCDET_SUSBHWDET		(3 << 0)
